### PR TITLE
feat: added mutect2 filtering rule

### PR DESCRIPTION
**Pull request by jonca79**
### This PR:

(If this is a release PR, no need to add following. Leave this part empty)
(Use the following lines to create a PR text body. Make sure to remove all non-relevant one after you're done)
(Repeat each field as many times as necessary)

Added: for new features.
Changed: for changes in existing functionality.
Deprecated: for soon-to-be removed features.
Removed: for now removed features.
Fixed: for any bug fixes.
Security: in case of vulnerabilities.

### Review and tests:
- [ ] Tests pass
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Code review
- [ ] `CHANGELOG.md` is updated
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: workflow/rules/gatk.smk [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        "{rule}: call variants in {input.map}"
+    wrapper:
+        "v1.3.1/bio/gatk/mutect"
+
+
+rule gatk_mutect2_gvcf:
+    input:
+        map="alignment/picard_mark_duplicates/{sample}_{type}_{chr}.bam",
+        bai="alignment/picard_mark_duplicates/{sample}_{type}_{chr}.bam.bai",
+        fasta=config["reference"]["fasta"],
```
Comment (marrip) [inline workflow/rules/gatk.smk:50]: probably better to use `get`

File: workflow/rules/gatk.smk [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+__copyright__ = "Copyright 2021, Patrik Smeds"
+__email__ = "patrik.smeds@scilifelab.uu.se"
+__license__ = "GPL-3"
+
+
+rule gatk_mutect2:
+    input:
+        map="alignment/picard_mark_duplicates/{sample}_{type}_{chr}.bam",
+        bai="alignment/picard_mark_duplicates/{sample}_{type}_{chr}.bam.bai",
+        fasta=config["reference"]["fasta"],
```
Comment (marrip) [inline workflow/rules/gatk.smk:11]: probably better to use `get`

File: workflow/rules/gatk.smk [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        bam=temp("snv_indels/gatk_mutect2/{sample}_{type}_{chr}.unfiltered.bam"),
+        bai=temp("snv_indels/gatk_mutect2/{sample}_{type}_{chr}.unfiltered.bai"),
+        stats=temp("snv_indels/gatk_mutect2/{sample}_{type}_{chr}.unfiltered.vcf.gz.stats"),
+        vcf=temp("snv_indels/gatk_mutect2/{sample}_{type}_{chr}.unfiltered.vcf.gz"),
+        tbi=temp("snv_indels/gatk_mutect2/{sample}_{type}_{chr}.unfiltered.vcf.gz.tbi"),
+        f1f2=temp("snv_indels/gatk_mutect2/{sample}_{type}_{chr}.unfiltered.f1r2.tar.gz"),
+    params:
+        extra=lambda wildcards: get_mutect2_extra(wildcards, "gatk_mutect2"),
+    log:
+        "snv_indels/gatk_mutect2/{sample}_{type}_{chr}.vcf.gz.log",
```
Comment (marrip) [inline workflow/rules/gatk.smk:23]: should be adjusted to new output name

File: workflow/rules/gatk.smk [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    benchmark:
+        repeat(
+            "snv_indels/gatk_mutect2/{sample}_{type}_{chr}.vcf.gz.benchmark.tsv",
```
Comment (marrip) [inline workflow/rules/gatk.smk:26]: same

File: workflow/rules/gatk.smk [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    wrapper:
+        "0.78.0/bio/gatk/mutect"
+
+
+rule gatk_mutect2_filter:
+    input:
+        vcf="snv_indels/gatk_mutect2/{sample}_{type}.merged.unfiltered.vcf.gz",
+        tbi="snv_indels/gatk_mutect2/{sample}_{type}.merged.unfiltered.vcf.gz.tbi",
+        stats="snv_indels/gatk_mutect2/{sample}_{type}.unfiltered.vcf.gz.stats",
+        ref=config["reference"]["fasta"],
```
Comment (marrip) [inline workflow/rules/gatk.smk:87]: better to use `get`

**Review by Smeds — APPROVED** [review]

File: workflow/rules/gatk.smk (hunk 3 above)
Comment (marrip) [reply workflow/rules/gatk.smk:23]: The vcf-output is called `snv_indels/gatk_mutect2/{sample}_{type}_{chr}.unfiltered.vcf.gz` thus log and benchmark should contain this as a prefix.

**Review by marrip — APPROVED** [review]

**Comment by marrip** [comment]
🍐  fect

**Pull request merged**